src/main/java/co/cask/wrangler/steps/Mask.java

```diff
@@ -0,0 +1,164 @@
+ *
+ * <p>
+ *  Substitution masking us generally used for masking credit card or SSN numbers.
+ *  This type of masking is fixed masking, where the pattern is applied on the
+ *  fixed length string.
+ *
+ *  <ul>
+ *    <li>Use of # will include the digit from the position.</li>
+ *    <li>Use x to mask the digit at that position</li>
+ *    <li>Any other characters will be inserted as-is.</li>
+ *  </ul>
+ *
+ *  <blockquote>
+ *    <pre>
+ *        Step step = new Mask(lineno, line, "ssn", "XXX-XX-####", 1);
+ *        Step step = new Mask(lineno, line, "amex", "XXXX-XXXXXX-X####", 1);
+ *    </pre>
+ *  </blockquote>
+ * </p>
+ *
+ * <p>
+ *   Fixed length shuffle masking performs obfuscation by using random character
+ *   substitution method. The data is randomly shuffled in the column.
+ *
+ *   <blockquote>
+ *     <pre>
+ *       Step step = new Mask(lineno, line, "150 Mars Avenue, Marcity, Mares", 2);
+ *     </pre>
+ *   </blockquote>
+ * </p>
+ */
+public class Mask extends AbstractStep {
+  private static final Logger LOG = LoggerFactory.getLogger(Mask.class);
+
+  // Specifies types of mask
+  public static final int MASK_NUMBER = 1;
+  public static final int MASK_SHUFFLE = 2;
+
+  // Masking pattern
+  private final String mask;
+
+  // Column on which to apply mask.
+  private final String column;
+
+  // Type of mask.
+  private final int maskType;
+
+  public Mask(int lineno, String detail, String column, int maskType) {
+    super(lineno, detail);
+    this.mask = "";
+    this.column = column;
+    this.maskType = maskType;
+  }
+
+  public Mask(int lineno, String detail, String column, String mask, int maskType) {
+    super(lineno, detail);
+    this.mask = mask;
+    this.column = column;
+    this.maskType = maskType;
+  }
+
+  /**
+   * Masks the column specified using either substitution method or shuffling.
+   *
+   * @param row Input {@link Row} to be wrangled by this step.
+   * @return A newly transformed {@link Row} with masked column.
+   * @throws StepException thrown when there is issue with masking
+   * @throws SkipRowException thrown when the row needs to be skipped
+   */
+  @Override
+  public Row execute(Row row) throws StepException, SkipRowException {
+    Row masked = new Row(row);
+    int idx = row.find(column);
+    if (idx != -1) {
+      if (maskType == MASK_NUMBER) {
+        try {
+          masked.setValue(idx, maskNumber((String)row.getValue(idx), mask));
+        } catch (Exception e) {
+          System.out.println(e.getMessage());
+        }
+      } else if (maskType == MASK_SHUFFLE) {
+        masked.setValue(idx, maskShuffle((String)row.getValue(idx), 0));
+      }
+    } else {
+      throw new StepException(toString() + " : '" +
+                                column + "' column is not defined. Please check the wrangling step."
+      );
+    }
+    return masked;
+  }
+
+  private String maskNumber(String number, String mask) {
```

Comment:
Yeah like credit cards, SSN or things like that. What would you suggest I name this ? @abhinavbansal246